integration-tests/bats/sql-remotes-server.bats

```diff
@@ -0,0 +1,556 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+remotesrv_pid=
+setup() {
+    setup_common
+    cd $BATS_TMPDIR
+    mkdir remotes-$$
+    mkdir remotes-$$/empty
+    echo remotesrv log available here $BATS_TMPDIR/remotes-$$/remotesrv.log
+    remotesrv --http-port 1234 --dir ./remotes-$$ &> ./remotes-$$/remotesrv.log 3>&- &
+    remotesrv_pid=$!
+    cd dolt-repo-$$
+    mkdir "dolt-repo-clones"
+}
+
+teardown() {
+    teardown_common
+    kill $remotesrv_pid
+    rm -rf $BATS_TMPDIR/remotes-$$
+}
+
+@test "sql-remotes-server: push with only one argument" {
+    skip "todo implement SQL push, pull, fetch"
+    dolt remote add test-remote http://localhost:50051/test-org/test-repo
+    run dolt sql -q "select dolt_push('test-remote')"
+    [ "$status" -eq 1 ]
+}
+
+@test "sql-remotes-server: push and pull master branch from a remote" {
```

Comment:
Isolate the SQL push and pull aspects. Fine to have a test that does both, but you definitely wants tests that only exercise one function (especially at this stage).

So do setup with the CLI, then make assertions about the output of the SQL func.

Your tests below already do this, they just need tweaked names and remove TODO and commented lines